File: converter-dotnet-core-console-vb/EventHandlingConversion.vb
Imports ceTe.DynamicPDF.Conversion

Class EventHandlingConversion

    Public Shared Sub Run()
        EventExcelExample("event-excel-output.pdf")
    End Sub

    Public Shared Sub EventWordExample(outputPdf As String)
        Dim word As New WordConverter(Program.GetResourcePath("DocumentA.doc"))
        AddHandler word.ProgressChanged, AddressOf converter_progresschanged
        AddHandler word.Completed, AddressOf converter_completed
        word.Convert(Program.GetOutputDocPath(outputPdf))
    End Sub

    Public Shared Sub EventExcelExample(outputPdf As String)
        Dim excel As New ExcelConverter(Program.GetResourcePath("DocumentA.xlsx"))
        AddHandler excel.DocumentParsing, AddressOf converter_documentparsing
        AddHandler excel.ProgressChanged, AddressOf converter_progresschanged
        AddHandler excel.Completed, AddressOf converter_completed
        excel.Convert(Program.GetOutputDocPath(outputPdf))
    End Sub

    Public Shared Sub converter_progresschanged(sender As Object, e As ProgressChangedEventArgs)
        Console.WriteLine(e.Status)
    End Sub

    Private Shared Sub converter_documentparsing(sender As Object, e As ExcelDocumentParsingEventArgs)
        Console.WriteLine("parsing spreadsheet")

        For Each excelWorksheet As ExcelWorksheet In e.Worksheets
            Console.WriteLine("Parsing a worksheet")
        Next
    End Sub

    Public Shared Sub converter_completed(sender As Object, e As ConversionCompletedEventArgs)
        If e.FailException IsNot Nothing Then
            Console.WriteLine("Conversion failed with:" & e.FailException.Message)
        Else
            Console.WriteLine("Conversion completed")
        End If
    End Sub

End Class

File: converter-dotnet-core-console-cs/EventHandlingConversion.cs

using ceTe.DynamicPDF.Conversion;
using System;

namespace converter_dotnet_core_console_cs
{
    class EventHandlingConversion
    {
        public static void Run()
        {
            EventExcelExample(@"event-excel-output.pdf");
        }


        public static void EventWordExample(string outputPdf)
        {
            WordConverter word = new WordConverter(Program.GetResourcePath(@"DocumentA.doc"));
            word.ProgressChanged += new ProgressChanged(converter_progresschanged);
            word.Completed += new ConversionCompleted(converter_completed);
            word.Convert(Program.GetOutputDocPath(outputPdf));
        }

        public static void EventExcelExample(string outputPdf)
        {
            ExcelConverter excel = new ExcelConverter(Program.GetResourcePath(@"DocumentA.xlsx"));
            excel.DocumentParsing += new ExcelDocumentParsing(converter_documentparsing);
            excel.ProgressChanged += new ProgressChanged(converter_progresschanged);
            excel.Completed += new ConversionCompleted(converter_completed);
            excel.Convert(Program.GetOutputDocPath(outputPdf));
        }


        public static void converter_progresschanged(object sender, ProgressChangedEventArgs e)
        {
            Console.WriteLine(e.Status);
        }

        static void converter_documentparsing(object sender, ExcelDocumentParsingEventArgs e)
        {
            Console.WriteLine("parsing spreadsheet");

            foreach (ExcelWorksheet excelWorksheet in e.Worksheets)
            {
                Console.WriteLine("Parsing a worksheet");
            }
        }

        public static void converter_completed(object sender, ConversionCompletedEventArgs e)
        {
            if (e.FailException != null)
                Console.WriteLine("Conversion failed with:" + e.FailException.Message);
            else Console.WriteLine("Conversion completed");
        }

    }
}
